viewing_party/main.py

```diff
@@ -0,0 +1,193 @@
+        return 0.0 
+
+    average_rating = sum_counter/len(user_data["watched"])
+    return average_rating 
+
+    # PROBLEM TWO 
+    # iterating through the watched list
+    # each loop gives us one movie 
+    # get genre from movie 
+    # add new genre into the dictionary as a key with count being the value, (check if genre is in the dict and add one) OR (add too dict at 1)
+    # Now iterate over different_genre dict
+    # Find highest value  
+    # return key to highest value 
+def get_most_watched_genre(user_data):
+    if not user_data["watched"]:
+        return None
+
+    genres = {}
+    for movie in user_data["watched"]:
+        for i, x in movie.items():
+            if i == "genre":
+                if x in genres:
+                    genres[x] += 1 
+                else:
+                    genres[x] = 1
+
+    count = 0
+    most_watched = ""
+
+    for i, x in genres.items():
+        if x > count:
+            count = x
+            most_watched = i 
+        
+    return most_watched
+
+
+
+
+# WAVE THREE 
+
+    # PROBLEM ONE 
+def get_unique_watched(user_data):
+    unique_movies = []
+    have_seen = {}
+    friends = user_data["friends"] # list of friends 
+    for movie in user_data["watched"]:
+        have_seen[movie["title"]] = False 
+    
+    for friend in friends:
+        friend_seen_list = friend["watched"]
+        for friend_movie in friend_seen_list:
+            friend_movie_title = friend_movie["title"]
+            if friend_movie_title in have_seen:
+                have_seen[friend_movie_title] = True
+
+    for movie in user_data["watched"]:
+        if not have_seen[movie["title"]]:
+            unique_movies.append(movie)
+
+    return unique_movies 
+
+
+    # PROBLEM TWO
+    # make an empty list user_has_not_seen_but_friends_have
+    # I need to iterate through the user watched list and compare the list to the movies that their friends have watched and identify which movies have been not been seen by the user
+    # add those unseen movies to the empty list user_has_not_seen_but_friends_have
+    # then return user_has_not_seen_but_friends_have
+def get_friends_unique_watched(user_data):
+    user_has_not_seen_but_friends_have = []
+    user_has_seen = user_data["watched"]
+    friends = user_data["friends"]
+    have_seen = {}
+
+    for friend in friends:
+        friend_seen_list = friend["watched"]
+        for friend_movie in friend_seen_list:
+            friend_movie_title = friend_movie["title"]
+            have_seen[friend_movie_title] = False 
+    for movie in user_has_seen:
+        if movie["title"] in have_seen:
+            have_seen[movie["title"]] = True 
+
+    for friend in friends:
+        friend_seen_list = friend["watched"]
+        for friend_movie in friend_seen_list:
+            if not have_seen[friend_movie["title"]]:
+                user_has_not_seen_but_friends_have.append(friend_movie)
+                have_seen[friend_movie["title"]] = True 
```

Comment:
Am I correct that this line `have_seen[friend_movie["title"]] = True ` is keeping you from having duplicates in your list. Another way you may consider achieving this result is through a compound conditional on line 135:  `if not have_seen[friend_movie["title"]] and friend_movie not in user_has+not_seen_but_friends_have`: